### PR TITLE
Fix Api Representation for media

### PR DESCRIPTION
**Pull request by quaxsze**
Fix https://github.com/etalab/data.gouv.fr/issues/985

### CODE REVIEW

File: udata_front/api.py [hunk 1]
```diff
@@ -77,11 +77,15 @@ def get(self):
...
-        return req.content
 
-    @apiv2.representation('image/*')
-    @apiv2.representation('audio/*')
-    def media(data, code, headers):
-        resp = make_response(bytes(data), code)
-        resp.headers.extend(headers)
-        return resp
+        values_gen = request.accept_mimetypes.values()
+        result_image = next((True for value in values_gen if 'image/' in value), False)
```
Comment (maudetes) [inline udata_front/api.py:82]: Could we do something like:
`any('image/' in value for value in values_gen)`

File: udata_front/api.py [hunk 2]
```diff
@@ -77,11 +77,15 @@ def get(self):
...
+        values_gen = request.accept_mimetypes.values()
+        result_sound = next((True for value in values_gen if 'audio/' in value), False)
+
+        if result_image or result_sound:
+            resp = make_response(bytes(req.content))
+            resp.headers["Content-Type"] = "image/*" if result_image else "audio/*"
+            return resp
```
Comment (maudetes) [inline udata_front/api.py:89]: If we return a content with `make_response`, it doesn't get process anymore by flask right?

**Review by maudetes — COMMENTED** [review]
Don't forget the changelog?

File: udata_front/api.py (hunk 2 above)
Comment (quaxsze) [reply udata_front/api.py:88]: Yes :)

File: udata_front/api.py [hunk 3]
```diff
@@ -77,11 +77,14 @@ def get(self):
...
         request.accept_mimetypes = MIMEAccept(accept)
-        return req.content
 
-    @apiv2.representation('image/*')
-    @apiv2.representation('audio/*')
-    def media(data, code, headers):
-        resp = make_response(bytes(data), code)
-        resp.headers.extend(headers)
-        return resp
+        values_gen = list(request.accept_mimetypes.values())
```
Comment (maudetes) [inline udata_front/api.py:81]: Not sure we need to meddle with `request.accept_mimetypes` anymore.
Could we directly use `make_response` based on `args['get']` being image or audio?

**Review by maudetes — COMMENTED** [review]
Seems to work perfectly, but I'm making a small suggestion

**Comment by quaxsze** [comment]
@nicolaskempf57 We remove the `api.representation` line for `media`. The whole APIv2 was passing through it.
Instead we use the `get` argument to know which header to include in the response. Does it look okey to you?

**Comment by nicolaskempf57** [comment]
> @nicolaskempf57 We remove the `api.representation` line for `media`. The whole APIv2 was passing through it. Instead we use the `get` argument to know which header to include in the response. Does it look okey to you?

Yes ! This was done because the captcha calls doesn't include any `Accept` header so flask-rest-x treats it as `application/json` and that prevent it for sending the response as bytes.

**Review by maudetes — APPROVED** [review]
Great! :clap: 

Don't forget to update the tests

**Review by nicolaskempf57 — APPROVED** [review]

**Pull request merged**